File: sandbox/play-core-mechanics/manage-character-sheet/create-character/create_character_stories.py
"""
# @toolset-manifest python -m tools manifest stories.stories:Stories
"""

"""Story data — regeneratable. Do not add logic or imports."""

from __future__ import annotations

from typing import Final


CREATE_CHARACTER: Final = {
    "story": "Create Character",
    "actor": "Player",
    "domain_terms": ("Character", "Abilities", "Ability", "Rank"),
    "evidence": (
        "sandbox/character/.context/module-context.md",
        "exploration example: all eight Abilities at rank 0",
    ),
    "main_flow": {
        "name": "new character has handbook abilities at rank zero",
        "given": ("no Character yet",),
        "interactions": (
            {
                "when": ("the Player creates a Character",),
                "then": (
                    "a Character exists with Abilities",
                    "And each of the eight Abilities has rank 0",
                ),
            },
        ),
    },
}

File: sandbox/play-core-mechanics/manage-character-sheet/create-character/create_character_stories.js
/**
 * @type {import('../../../../stories/code/javascript/seeds/story-types').Story}
 */
export const CreateCharacter = {
  story:       'Create Character',
  actor:       'Player',
  domainTerms: ['Character', 'Abilities', 'Ability', 'Rank'],
  evidence:    ['sandbox/character/.context/module-context.md', 'exploration example: all eight Abilities at rank 0'],

  newCharacterHasHandbookAbilitiesAtRankZero: {
    name: 'new character has handbook abilities at rank zero',
    given: [
      'no Character yet',
    ],
    interactions: [
      {
        when: [
          'the Player creates a Character',
        ],
        then: [
          'a Character exists with Abilities',
          'And each of the eight Abilities has rank 0',
        ],
      },
    ],
  },
}

export const storyNames = ['Create Character']
